### ISSUE TITLE
After reading in .h5ad: `pp.highly_variable_genes` with `flavor='seurat_v3'` expects raw count data

### ISSUE THREAD

**Issue by jtabin** [scverse/scanpy#1782]
- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of scanpy.
---

Hello, I am trying to run `sc.pp.highly_variable_genes` with `flavor='seurat_v3'` on some data, but it is giving me an error. The data has been run through Kallisto Bustools and is being loaded in as an .h5ad file; it should be raw count data. 

### Minimal code sample

```python
zf_48 = anndata.read_h5ad("data.h5ad")
zf_48.var["gene_ids"] = zf_48.var.index.values

t2g = pd.read_csv("transcripts_to_genes_no_rm.txt", header=None, names=["tid", "gene_id", "gene_name"], sep="\t")
t2g.index = t2g.gene_id
t2g = t2g.loc[~t2g.index.duplicated(keep='first')]
zf_48.var["gene_name"] = zf_48.var.gene_ids.map(t2g["gene_name"])
zf_48.var.index = zf_48.var["gene_name"] 

zf_48.var_names_make_unique()
sc.pp.filter_cells(zf_48, min_genes=550)
sc.pp.filter_genes(zf_48, min_cells=10)
zf_48
#AnnData object with n_obs × n_vars = 887 × 13180
#    obs: 'n_genes'
#   var: 'gene_name', 'gene_ids', 'n_cells'

zf_48.var['mt'] = zf_48.var_names.str.startswith('mt-') 
sc.pp.calculate_qc_metrics(zf_48, qc_vars=['mt'], percent_top=None, log1p=False, inplace=True)
zf_48 = zf_48[zf_48.obs.n_genes_by_counts < 2500, :]
zf_48 = zf_48[zf_48.obs.pct_counts_mt < 10, :]
sc.pp.highly_variable_genes(zf_48, flavor='seurat_v3', span=1)
```
Here is the error:
```pytb
ValueError                                Traceback (most recent call last)
<ipython-input-170-37cd37b7326e> in <module>
     16 zf_48 = zf_48[zf_48.obs.n_genes_by_counts < 2500, :]
     17 zf_48 = zf_48[zf_48.obs.pct_counts_mt < 10, :]
---> 18 sc.pp.highly_variable_genes(zf_48, flavor='seurat_v3', span=1)

~/opt/anaconda3/lib/python3.8/site-packages/scanpy/preprocessing/_highly_variable_genes.py in highly_variable_genes(adata, layer, n_top_genes, min_disp, max_disp, min_mean, max_mean, span, n_bins, flavor, subset, inplace, batch_key)
    413 
    414     if flavor == 'seurat_v3':
--> 415         return _highly_variable_genes_seurat_v3(
    416             adata,
    417             layer=layer,

~/opt/anaconda3/lib/python3.8/site-packages/scanpy/preprocessing/_highly_variable_genes.py in _highly_variable_genes_seurat_v3(adata, layer, n_top_genes, batch_key, span, subset, inplace)
     59     X = adata.layers[layer] if layer is not None else adata.X
     60     if check_nonnegative_integers(X) is False:
---> 61         raise ValueError(
     62             "`pp.highly_variable_genes` with `flavor='seurat_v3'` expects "
     63             "raw count data."

ValueError: `pp.highly_variable_genes` with `flavor='seurat_v3'` expects raw count data.
```

Am I loading the data in wrong? This processing has worked for data loaded in using 'sc.read_10x_mtx()'.

#### Versions

<details>

-----
anndata     0.7.5
scanpy      1.6.0
sinfo       0.3.1
-----
PIL                 8.0.1
PyObjCTools         NA
anndata             0.7.5
anndata2ri          1.0.5
appnope             0.1.2
attr                20.3.0
backcall            0.2.0
bottleneck          1.3.2
cffi                1.14.4
cloudpickle         1.6.0
colorama            0.4.4
cycler              0.10.0
cython_runtime      NA
cytoolz             0.11.0
dask                2020.12.0
dateutil            2.8.1
decorator           4.4.2
get_version         2.1
h5py                3.1.0
idna                2.10
igraph              0.8.3
ipykernel           5.4.2
ipython_genutils    0.2.0
ipywidgets          7.5.1
jedi                0.17.2
jinja2              2.11.2
joblib              1.0.0
jsonschema          3.2.0
kiwisolver          1.3.1
legacy_api_wrap     1.2
leidenalg           0.8.3
llvmlite            0.35.0
louvain             0.7.0
markupsafe          1.1.1
matplotlib          3.3.3
mpl_toolkits        NA
natsort             7.1.0
nbformat            5.0.8
numba               0.52.0
numexpr             2.7.1
numpy               1.19.4
packaging           20.8
pandas              1.1.5
parso               0.7.0
pexpect             4.8.0
pickleshare         0.7.5
pkg_resources       NA
prometheus_client   NA
prompt_toolkit      3.0.8
psutil              5.7.3
ptyprocess          0.6.0
pvectorc            NA
pygments            2.7.3
pyparsing           2.4.7
pyrsistent          NA
pytz                2020.4
rpy2                3.3.6
scanpy              1.6.0
scipy               1.5.4
seaborn             0.11.0
send2trash          NA
setuptools_scm      NA
sinfo               0.3.1
six                 1.15.0
sklearn             0.23.2
skmisc              0.1.3
sphinxcontrib       NA
statsmodels         0.12.1
storemagic          NA
tables              3.6.1
tblib               1.7.0
terminado           0.9.1
texttable           1.6.3
tlz                 0.11.0
toolz               0.11.1
tornado             6.1
traitlets           5.0.5
typing_extensions   NA
tzlocal             NA
wcwidth             0.2.5
yaml                5.3.1
zmq                 20.0.0
zope                NA
-----
IPython             7.19.0
jupyter_client      6.1.7
jupyter_core        4.7.0
jupyterlab          2.2.9
notebook            6.1.5

</details>

**Comment by ivirshup** [comment]
In the current release, we check for the counts being integer valued. kallisto can assign partial counts, (e.g a gene can have 1.5 counts) which triggers the check, triggering an error. For the next bugfix release we've softened consequences of this check failing to a warning, and the check can be skipped.

See discussion in #1642 and #1679 for details.

**Comment by jtabin** [comment]
Thank you!

**Comment by ivirshup** [comment]
Should be fixed as of `1.7.2`, which was just released.
